x-pack/plugins/entity_manager/server/lib/entities/ingest_pipeline/generate_latest_processors.ts

```diff
@@ -156,6 +139,55 @@ export function generateLatestProcessors(definition: EntityDefinition) {
         value: definition.identityFields.map((identityField) => identityField.field),
       },
     },
+    {
+      script: {
+        description: 'Generated the entity.id field',
+        source: cleanScript(`
+        // This function will recursively collect all the values of a HashMap of HashMaps
+        Collection collectValues(HashMap subject) {
+          Collection values = new ArrayList();
+          // Iterate through the values
+          for(Object value: subject.values()) {
+            // If the value is a HashMap, recurse
+            if (value instanceof HashMap) {
+              values.addAll(collectValues((HashMap) value));
+            } else {
+              values.add(String.valueOf(value));
+            }
+          }
+          return values;
+        }
+
+        // Create the string builder
+        StringBuilder entityId = new StringBuilder();
+
+        if (ctx["entity"]["identity"] != null) {
+          // Get the values as a collection
+          Collection values = collectValues(ctx["entity"]["identity"]);
+
+          // Convert to a list and sort
+          List sortedValues = new ArrayList(values);
+          Collections.sort(sortedValues);
+
+          // Create comma delimited string
+          for(String instanceValue: sortedValues) {
+            entityId.append(instanceValue);
+            entityId.append(":");
+          }
+
+            // Assign the entity.id
+          ctx["entity"]["id"] = entityId.length() > 0 ? entityId.substring(0, entityId.length() - 1) : "unknown";
+        }
+       `),
+      },
+    },
+    {
+      fingerprint: {
+        fields: ['entity.id'],
+        target_field: 'entity.id',
+        method: 'MurmurHash3',
+      },
+    },
```

Comment:
@tommyers-elastic I kind of wonder if we shouldn't take this opportunity to simplify the `entity.id` generation? I purpose we remove the hashing AND change the `entity.id` to be:
```
{
  set: {
     field: 'entity.id',
     value: definition.identityFields.map((identityField) => `{{entity.identity.${identityField.field)}}`).join(':'),
  }
}
```
For something like `['service.name', 'service.environment']` it would look like `{{entity.identity.service.name}}:{{entity.identity.service.environment}}`

Thoughts @miltonhultgren @klacabane

Reply:
This makes sense and it is implemented in https://github.com/elastic/kibana/pull/193652. I'd love to keep these separate as this PR has been opened for a while now and all the lights are finally green

Reply:
➕ ➕  lets get this merged!

Reply:
MERGE IT!

<img width="489" alt="image" src="https://github.com/user-attachments/assets/76785230-91fa-4894-9ae0-b80005d59674">